src/borg/testsuite/__init__.py

```diff
@@ -235,26 +235,63 @@ def _assert_dirs_equal_cmp(self, diff, ignore_bsdflags=False, ignore_xattrs=Fals
             self._assert_dirs_equal_cmp(sub_diff, ignore_bsdflags=ignore_bsdflags, ignore_xattrs=ignore_xattrs, ignore_ns=ignore_ns)
 
     @contextmanager
-    def fuse_mount(self, location, mountpoint, *options):
-        os.mkdir(mountpoint)
-        args = ['mount', location, mountpoint] + list(options)
-        self.cmd(*args, fork=True)
+    def fuse_mount(self, location, mountpoint=None, *options, **kwargs):
+        if mountpoint is None:
+            mountpoint = tempfile.mkdtemp()
+        else:
+            os.mkdir(mountpoint)
+        if 'fork' not in kwargs:
+            # For a successful mount, `fork = True` is required. The
+            # option to run the command with `fork = False` is still
+            # necessary to test mount failures like attempting to mount
+            # a read-only repo.
+            kwargs['fork'] = True
+        self.cmd('mount', location, mountpoint, *options, **kwargs)
         self.wait_for_mount(mountpoint)
         yield
         umount(mountpoint)
+        self.wait_for_mount(mountpoint, umount=True)
         os.rmdir(mountpoint)
         # Give the daemon some time to exit
-        time.sleep(.2)
+        time.sleep(0.2)
+
+    def wait_for_mount(self, mountpoint, timeout=5, umount=False):
+        """Wait until a path meets specified mount point status
 
-    def wait_for_mount(self, path, timeout=5):
-        """Wait until a filesystem is mounted on `path`
+        By default, checks if `mountpoint` is a mount point. To check
+        if `mountpoint` is *not* a mount point, set `umount = True`.
         """
         timeout += time.time()
         while timeout > time.time():
-            if os.path.ismount(path):
+            if os.path.ismount(mountpoint) != umount:
                 return
-            time.sleep(.1)
-        raise Exception('wait_for_mount(%s) timeout' % path)
+            time.sleep(0.1)
+        message = 'Waiting for %s of %s' % ('umount' if umount else 'mount', mountpoint)
+        raise TimeoutError(message)
```

Comment:
I had to modify `wait_for_mount` a little bit, too. It's now possible to wait for mounting as well as unmounting. I also replaced the generic `Exception` with `TimeoutError` which seems more suitable.

Reply:
somehow a signature like this would be more self-explaining:
```
wait_for_mountstate(mountpoint, *, mounted, timeout=5)
```

calls:
```
wait_for_mountstate('/mnt', mounted=True)
wait_for_mountstate('/mnt', mounted=False)
```

Reply:
I agree with your comment. Did you mark this as resolved by accident?

Reply:
it doesn't show as resolved here.

Reply:
Only because I clicked on "unresolve" before commenting. :wink:  (I wasn't sure if you'd get a notification for a comment in a resolved thread.)

I'll change the function. :+1: